core/src/main/java/org/apache/carbondata/core/constants/CarbonCommonConstants.java

```diff
@@ -1519,6 +1519,12 @@ private CarbonCommonConstants() {
 
   public static final String CARBON_QUERY_PREFETCH_ENABLE_DEFAULT = "true";
 
+  @CarbonProperty
+  public static final String CARBON_QUERY_STAGE_INPUT =
+      "carbon.query.stage.input.enable";
+
+  public static final String CARBON_QUERY_STAGE_INPUT_DEFAULT = "false";
```

Comment:
default should be `true` ? because it can help avoid latency by default

Reply:
I'd like to make it disable by default because it may impact the query performance. Stage files are not sorted and need to be scanned